### ISSUE TITLE
Fully support BaseEstimator

### ISSUE THREAD

**Issue by akatief** [UKPLab/supervised-multidimensional-scaling#103]
BaseEstimator is a sklearn base class that allows interfacing with some of its functionalities like the Pipeline class.

Currently, SMDS is a BaseEstimator but fails the checks. In particular, SupervisedMDS should not initialize any property other than init arguments. See scikit-learn Developing scikit-learn estimators for further details. A change in score function might also be needed to uniform with sklearn interface by dropping the metric argument. If that's the case, default to SCALE_NORMALIZED_STRESS.
